Framework/API/inc/MantidAPI/Algorithm.h

```diff
@@ -189,6 +193,163 @@ class MANTID_API_DLL Algorithm : public IAlgorithm,
   /// Override if the algorithm is not part of the Mantid distribution.
   const std::string helpURL() const override { return ""; }
 
+  template <typename T> void checkWorkspaceType() const {
```

Comment:
Hey sorry about that, this wasn't meant for review yet. There are a few things like that which need be changed (copied and pasted from other code).